src/lib/support/Pool.h

```diff
@@ -331,6 +388,86 @@ class HeapObjectPool : public internal::Statistics, public HeapObjectPoolExitHan
 #endif // __SANITIZE_ADDRESS__
     }
 
+    /// Provides iteration over active objects in the pool.
+    ///
+    /// NOTE: There is extra logic to allow objects release WHILE the iterator is
+    ///       active while still allowing to advance the iterator.
```

Comment:
Having heap pools support that but non-heap pools not support it sounds like a recipe for bugs where people test with one pool type and then it fails with the other.

Reply:
Both pools support it - that was the point of this comment. I would like to eventually replace the `ForEachActiveObject` with the iterators, hence this support ... otherwise I would have dropped it. I believe this extra logic is a pain.

Reply:
Updated the bitmap comment. They were intended toboth allow similar functionality and not invalidate iterators.